src/compiler/semantic/rules.rs

```diff
@@ -251,10 +249,48 @@ fn redeclare_rule(
     }
 }
 
+// Cannot use true or false if it's not inside of a logic expression.
+fn true_false_rule(analyser: &mut Analyser, expr: &Expression<BigInt, Identifier>) {
+    use Expression::*;
```

Comment:
@rutefig I think this does not go recursively and hence only checks at the second level of the Expression AST

Reply:
True, fixing it